### PR TITLE
Fixing Vulnerability Detector unit tests. 

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#7926|
## Description
Adding and fixing UTs for the changes introduced as part of #7749 that affected the Vulnerability Detector module.
#### ctest
image
image
#### Valgrind
`valgrind -s --leak-check=full --show-leak-kinds=all --num-callers=20 --trace-children=yes --track-origins=yes`
##### Results
[valgrind.txt](https://github.com/wazuh/wazuh/files/6327035/valgrind.txt)
## Tests

- Compilation without warnings in every supported platform
  - [X] Linux
- [X] Source installation
- [X] Source upgrade
- [X] Review logs syntax and correct language

- Memory tests for Linux
  - [X] Valgrind (memcheck and descriptor leaks check)

- [X] Added unit tests (for new features)

### CODE REVIEW

File: src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c [hunk 1]
```diff
@@ -224,6 +220,10 @@ static int build_test_cve_report(vu_report* report, int add_title, int add_condi
     report->ref_sources[1] = NULL;
     // Reported software
     os_strdup("libhogweed4", report->software);
+    if (1 == add_title){
```
Comment (DProvinciani) [inline src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c:223]: Not sure whether this is still necessary, as well as the `add_title`. I took a quick look at the code and the function `build_test_cve_report` is always called with the first parameter as `0`.

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Please, apply changes according to the comments.

**Review by palaciosjeremias — APPROVED** [review]
LGTM

File: src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c (hunk 1 above)
Comment (MiguelazoDS) [reply src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c]: You're right we don't need this anymore.

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**